src/nashEquilibria.jl

```diff
@@ -0,0 +1,207 @@
+    for i in 1:numPlayers
+         minArray[i] = minimum(g.players[i].payoff_array)
+    end
+    normalizer = abs(minimum(minArray)) + 1
+
+    # Alter the payoff matrices.
+    for i in 1:numPlayers
+        g.players[i].payoff_array += normalizer
+    end
+    return g
+end
+
+# This is currently very slow; i.e., each call removes one dominated strategy for one player.
+
+"""
+Simplify a n-player normal form game by the iterated removal of dominated strategies.
+
+##### Arguments
+
+- `g::NormalFormGame` : A NormalFormGame object.
+
+##### Returns
+
+- `g::NormalFormGame` : A NormalFormGame object.
+"""
+function unDominate!(g::NormalFormGame)
+    changeFlag = true
+    changesMade = false
+    while changeFlag == true
+        changeFlag = unDominate_Auxiliary!(g::NormalFormGame)
+        changesMade = changeFlag
+    end
+    return changesMade
+end
+
+function unDominate_Auxiliary!(g::NormalFormGame)
+    # Iterate over players, flagging strictly dominated strategies, and then eliminating.
+    changesMade = false
+    for i in 1:num_players(g)
+        player = g.players[i]
+        numActions = num_actions(player)
+        payoffMatrix = player.payoff_array
+        dominatedActions = Array(Integer,0)
+        for j in 1:numActions
+            currentAction = payoffMatrix[j,:]
+            for k in symdiff(numActions,j)
+                 otherAction = payoffMatrix[k,:]
+                 if size(otherAction.>=currentAction) == size(currentAction)
+                      g.players[i].payoff_array = g.players[i].payoff_array[symdiff(numActions,j),:]
+                      changesMade = true
+                      return changesMade
+                 end
+            end
+        end
+    end
+end
+
+# Auxiliary function for Lemke-Howson.
+function constructTableaux(g::NormalFormGame)
+    # Check params.
+    if num_players(g) != 2
+         error("Two-player game required.")
+    end
+
+    # Bring in primitives.
+    A = g.players[1].payoff_array
+    B = g.players[2].payoff_array
+    m = num_actions(g.players[1])
+    n = num_actions(g.players[2])
+
+    # Create tableaux.
+    tableaux = Array(Matrix,2)
+    tableaux[1] = cat(2, B, eye(n), ones(n, 1))
+    tableaux[2] = cat(2, eye(m), A, ones(m, 1))
+
+    return tableaux
+end
+
+# Testing implementing using another package. Need to look at syntax for passing in maxPivots.
+
+"""
+Apply the Lemke-Howson algorithm to a normal form game.
+
+##### Arguments
+
+- `g::NormalFormGame` : A NormalFormGame object.
+- `maxPivots::Integer` : A bound on the maximum number of iterations.
+
+##### Returns
+
+- `nashEquilibrium::Tuple` : A Nash equilibrium for the game, given as a tuple of MixedAction objects.
+
+"""
+function applyLH(g::NormalFormGame, maxPivots::Integer=50000)
+     # Pre-process.
+     unDominate!(g)
+     unDegenerate!(g)
+     tableaux = constructTableaux(g)
+
+     # Solve.
+     nashEquilibrium = nonneg_lsq(tableaux[1],tableaux[2];alg=:pivot)
```

Comment:
can just return from this line, no need to assign it to an intermediate local variable and then just return that variable